adhocteam/Head-Start-TTADP
# CI: Cleanup docker-executor, split lint_and_audit job

**Description of change**
Several small clean up tasks:
1. Creates plain `docker-executor` that doesn't include postgres image. Uses this executor for build and deploy steps because there we don't actually need postgres. Though the postgres image step is run async and wasn't technically holding any jobs, it was still an unnecessary step that adds visual clutter and makes our workflow times look higher than they are.
2. Removes `EXIT_CODE` evaluation from `yarn audit` ([discussed in slack](https://gsa-tts.slack.com/archives/C017RJ4GEET/p1607629941472800?thread_ts=1607629127.468800&cid=C017RJ4GEET))
3. Split up `lint_and_audit` job. For this job, spinning up the environment and attaching the workspace takes longer than the actual linting and auditing steps. We're not saving any time by running this as its own job, particularly because the "fan out" section of our `build_test_deploy ` workflow has 6 jobs, which means we always max out our 4 job concurrency limit. Cutting up this fast/quick job means we're more likely to run our slow/big job (`dynamic_security_scan`) in the first concurrent batch.
4. `yarn <script>` instead of `yarn run <script>`

**How to test**
Check out this branch's [most recent workflow run in circleci](https://app.circleci.com/pipelines/github/adhocteam/Head-Start-TTADP?branch=sj-reduce-ci-jobs)

**Issue(s)**
* Cruft

**Checklist**

- [n/a] Meets issue criteria
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] Documentation updated - **_I think this is self documenting, let me know if you want more comments or something in the README_**
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
